### PR TITLE
Support customize common path in welcome theme

### LINKED ISSUE #22401: Common resources in Welcome page didn't resolve correctly
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

dist/quarkus

### Describe the bug

https://github.com/keycloak/keycloak/blob/8f38e269700694dc8a34a673038d12ec3f6108cc/quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/services/resources/QuarkusWelcomeResource.java#L185-L185

This should use `import` variable in theme.properties instead of hard-coded.

### Version

v22.0.1

### Expected behavior

correct common path

### Actual behavior

404 not found

### How to Reproduce?

use `resourcesCommonPath` in welcome page.

```html
<link href="${resourcesCommonPath}/img/favicon.ico" rel="icon"/>
```

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by kamontat**
fixes #22401

Because I don't want to introduce any break changes, so I just copied some code from UrlBean to handle common similar to other pages (e.g. login)

https://github.com/keycloak/keycloak/blob/580c5ca5bc410fca6875216572ce218436cb53e3/services/src/main/java/org/keycloak/forms/login/freemarker/model/UrlBean.java#L123-L135

### CODE REVIEW

**Comment by jonkoops** [comment]
@ssilvert I think this might be up your alley?

**Comment by ssilvert** [comment]
@kamontat Sorry for the lack of action on this one.  @jonkoops and I are trying to come up with a standard for dealing with "common" that will be consistent for all theme features (welcome, email, admin console, account console, login).

**Comment by kamontat** [comment]
If you need me, I happy to help :)

**Comment by ssilvert** [comment]
@jonkoops Given our recent conversations, I'm inclined to merge this as-is (with conflicts fixed, of course).  It's a little inconsistent with the other theme types, but the Welcome theme type is kind of on its own anyway.  So this seems to make sense.

@kamontat Can you fix the conflicts?

**Comment by kamontat** [comment]
@ssilvert Resolved conflict.

File: services/src/main/java/org/keycloak/services/resources/WelcomeResource.java [hunk 1]
```diff
@@ -189,16 +189,20 @@ private Response createWelcomePage(String successMessage, String errorMessage) {
                 return Response.status(302).location(adminUrl).build();
             }
 
-            Map<String, Object> map = new HashMap<>();
+            String commonPath = themeProperties.getProperty("common");
+            if (commonPath == null || commonPath.isEmpty()) {
+                commonPath = "common/keycloak";
+            }
```
Comment (jonkoops) [inline services/src/main/java/org/keycloak/services/resources/WelcomeResource.java:195]: I believe we can provide a default value to `getProperty()` here, it wouldn't handle the empty case, but that is a user-error anyways.
```suggestion
            String commonPath = themeProperties.getProperty("common", "common/keycloak");
```

**Review by jonkoops — CHANGES_REQUESTED** [review]

**Review by ssilvert — DISMISSED** [review]

**Review by jonkoops — DISMISSED** [review]

**Comment by jonkoops** [comment]
@kamontat can you squash your commits and sign off on them? Also please don't keep merging `main` into your branch unless there are conflicts, as it will generate additional notifications and CI runs. If you do have to get things from `main` I recommend that you use a rebase instead, as it keeps the GIT history clean.

**Comment by kamontat** [comment]
I make some mistake when rebase the commits, so I create new commit and force-push instead. This will resolve DCO as well. Sorry for any inconvenience.

**Review by jonkoops — APPROVED** [review]

**Review by ssilvert — APPROVED** [review]

**Pull request merged**